daemon/logger/ring.go

```diff
@@ -19,6 +19,7 @@ type RingLogger struct {
 	l         Logger
 	logInfo   Info
 	closeFlag int32
+	wg        *sync.WaitGroup
```

Comment:
Sorry I didn't catch this earlier, `sync.WaitGroup` should be used as a value here.

Then at line 45 there is no need to initialize it either.

Reply:
That is, not a pointer.

Reply:
Updated. Please check. 

FWIW, the sync.Waitgroup funcs `Add` and `Wait` [0] are on the pointer receiver, so I think both would have worked. Curious why you preferred to use it as a value.

[0] https://golang.org/pkg/sync/#WaitGroup.Add

Reply:
It's just the "normal" way to use it... or at least how we use it throughout the codebase.
Only take the pointer when you need to pass it to another function.

Reply:
Okay.. thank you for explaining. I have updated it.